lms/static/js/learner_dashboard/RecommendationsPanel.jsx

```diff
@@ -60,44 +62,57 @@ class RecommendationsPanel extends React.Component {
-          {this.state.coursesList.length ? this.state.coursesList.map(course => (
-            <a href={course.marketing_url} className="course-link"
-               onClick={() => this.onCourseSelect(course.course_key)}>
-              <div className="course-card box-shadow-down-1 bg-white mb-3">
-                <div className="box-shadow-down-1 image-box">
-                  <img
-                    className="panel-course-img"
-                    src={course.logo_image_url}
-                    alt="course image"
-                  />
-                </div>
-                <div className="course-title pl-3">
-                  {course.title}
-                </div>
-              </div>
-            </a>
-          )) : (
-            <div className="d-flex justify-content-center align-items-center">
+        {this.state.isLoading ? (
+          <div>
+            <div className="recommend-heading mb-4">{gettext('Recommendations for you')}</div>
+            <div className="d-flex justify-content-center align-items-center spinner-container">
               <div role="status" className="spinner">
                 <span className="sr-only">{gettext('loading')}</span>
               </div>
             </div>
-          )}
-        </div>
-
+          </div>
+        ) : (
+          <div>
+            {this.state.coursesList.length ? (
+              <div>
+                <div className="recommend-heading mb-4">{gettext('Recommendations for you')}</div>
+                <div>
+                  {this.state.coursesList.map(course => (
+                    <a href={course.marketing_url} className="course-link"
+                      onClick={() => this.onCourseSelect(course.course_key)}>
+                      <div className="course-card box-shadow-down-1 bg-white mb-3">
+                        <div className="box-shadow-down-1 image-box">
+                          <img
+                            className="panel-course-img"
+                            src={course.logo_image_url}
+                            alt="course image"
+                          />
+                        </div>
+                        <div className="course-title pl-3">
+                          {course.title}
+                        </div>
+                      </div>
+                    </a>
+                  ))}
+                </div>
+              </div>
+            ) : null}
+          </div>
+        )}
         {this.props.exploreCoursesUrl ? (
-          <div className="d-flex justify-content-center">
-            <a href={this.props.exploreCoursesUrl}
-               className="panel-explore-courses justify-content-center align-items-center">
-              {gettext('Explore courses')}
-              <span className="icon fa fa-search search-icon" aria-hidden="true"/>
-            </a>
+          <div>
+            {!(this.state.coursesList.length || this.state.isLoading) &&
```

Comment:
I think we should only show the text once the result from backend has no recommendations.
```suggestion
            {!(this.state.coursesList.length && this.state.isLoading) &&
```